[
  {
    "user": "U257C4K4N",
    "text": "<@U257C4K4N|jainshravan123> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472194871.000904",
    "id": "26be7372",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25808F8F",
    "text": "<@U25808F8F|stilett> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472197765.000905",
    "id": "26be7373",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U2584DXL4",
    "text": "<@U2584DXL4|viraj49> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472200438.000906",
    "id": "26be7374",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25AJ9X5X",
    "text": "<@U25AJ9X5X|gdaniels> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472216753.000907",
    "id": "26be7375",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25B3JQQ2",
    "text": "<@U25B3JQQ2|ikakus> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472218969.000908",
    "id": "26be7376",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25BF96SC",
    "text": "<@U25BF96SC|abarza> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472223130.000909",
    "id": "26be7377",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25BKCJJY",
    "text": "<@U25BKCJJY|karanganesan> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472224053.000910",
    "id": "26be7378",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25BKCJJY",
    "text": "Which is the best place to create Cross Platform apps ?",
    "ts": "1472224142.000911",
    "id": "26be7379",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25BKCJJY",
    "text": "I am a beginner ...",
    "ts": "1472224149.000912",
    "id": "26be7380",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25CTHVN0",
    "text": "<@U25CTHVN0|shivombhatt> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472229870.000913",
    "id": "26be7381",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25D5B5NZ",
    "text": "<@U25D5B5NZ|jamesgrich> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472230136.000914",
    "id": "26be7382",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U061HDV96",
    "text": "<@U25BKCJJY>: hi, welcome. Most of the stuff here is native Android development. I recommend you to go phonegap, xamarin, react if you really want to build a cross platform app. Just know that a cross platform app is never gonna be as native-looking and performant as a native app.",
    "ts": "1472234036.000915",
    "id": "26be7383",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25EM6WET",
    "text": "<@U25EM6WET|joem86> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472236946.000916",
    "id": "26be7384",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25ENCG0L",
    "text": "<@U25ENCG0L|ryez> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472239046.000917",
    "id": "26be7385",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HYKCG1",
    "text": "<@U25HYKCG1|danny> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472267817.000918",
    "id": "26be7386",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HYKCG1",
    "text": "Hello all, I am new to android development and had a question\nI am trying to show a toast message from a class that extends `FirebaseMessagingService` from the function `onMessageRecieved` , and I am facing a problem because I do not know how to get the `Context` of the top most activity, any  help will be greatly appreciated",
    "ts": "1472268068.000919",
    "id": "26be7387",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U1HM4MPLJ",
    "text": "You can use the service context itself",
    "ts": "1472269266.000920",
    "id": "26be7388",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U1HM4MPLJ",
    "text": "Or call `Service#getApplicationContext`",
    "ts": "1472269290.000921",
    "id": "26be7389",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U1HM4MPLJ",
    "text": "Internally, toasts make use of the application context from whatever context you supply anyway",
    "ts": "1472269399.000922",
    "id": "26be7390",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U1HM4MPLJ",
    "text": "<@U25HYKCG1> ^",
    "ts": "1472269410.000923",
    "id": "26be7391",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HYKCG1",
    "text": "<@U1HM4MPLJ> Thanks I will try it out",
    "ts": "1472269817.000924",
    "id": "26be7392",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "<@U25HPU938|tim_malseed> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472270109.000925",
    "id": "26be7393",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HQJX6J",
    "text": "<@U25HQJX6J|huuhoang> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472270676.000926",
    "id": "26be7394",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HQKKDL",
    "text": "<@U25HQKKDL|tien.workinfo> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472270701.000927",
    "id": "26be7395",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HP2W04",
    "text": "<@U25HP2W04|tam01235> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472271364.000928",
    "id": "26be7396",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HUGTE0",
    "text": "<@U25HUGTE0|akhilshukla> has joined the channel",
    "subtype": "channel_join",
    "ts": "1472276808.000929",
    "id": "26be7397",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "hello, can someone explain what checked and unchecked exceptions are visually. lots of thanks",
    "ts": "1472277945.000930",
    "id": "26be7398",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "<@U0G47NBQV> \n\nI don’t know if I can explain it visually, but think of it like this:\n\nA ‘checked’ exception is one that you have to deal with before your code will compile. It is ‘checked’ at compile time. If you don’t handle it (using a try-catch somewhere) then your code won’t compile.\n\nAn ‘unchecked’ exception is one that will still compile - you don’t have to deal with it in your code. But, if that exception occurs when the app is running, your app will crash.",
    "ts": "1472278151.000931",
    "id": "26be7399",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "Super easy",
    "ts": "1472278174.000932",
    "id": "26be7400",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "Create plain old java project",
    "ts": "1472278188.000933",
    "id": "26be7401",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "try to read a file",
    "ts": "1472278198.000934",
    "id": "26be7402",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "See what happens when you don’t add `throws` or `try catch`",
    "ts": "1472278235.000935",
    "id": "26be7403",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "that’s a checked exception",
    "ts": "1472278245.000936",
    "id": "26be7404",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "experiment 2",
    "ts": "1472278256.000937",
    "id": "26be7405",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "run this",
    "ts": "1472278263.000938",
    "id": "26be7406",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "<@U25HPU938> alright, I get the part of that checked exceptions should be dealt with.",
    "ts": "1472278273.000939",
    "id": "26be7407",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "`int[] myArr = new int[2];`\n`int[3] = 5;`",
    "edited": {
      "user": "U08GFT5V5",
      "ts": "1472278301.000000"
    },
    "ts": "1472278295.000940",
    "id": "26be7408",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "but can ‘unchecked` exceptions be handled as well?",
    "ts": "1472278309.000942",
    "id": "26be7409",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "Yes, definitely.",
    "ts": "1472278316.000943",
    "id": "26be7410",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "You get an exception, while it’s running",
    "ts": "1472278325.000944",
    "id": "26be7411",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "but doesn’t give an error at compile time",
    "ts": "1472278343.000945",
    "id": "26be7412",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "They can both be handled in exactly the same way. The only real difference is with ‘checked’ exception, you are forced to handle it or your code won’t compile.",
    "ts": "1472278348.000946",
    "id": "26be7413",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "More or less",
    "ts": "1472278357.000947",
    "id": "26be7414",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "okay got it.",
    "ts": "1472278453.000948",
    "id": "26be7415",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "So, look at <@U08GFT5V5>’s second example:\n\n```\nint[] myArr = new int[2];\nint[3] = 5;```\n\nWe know this will crash, because the int[] only has a size of 2, and we’re trying to add something to int[3]. But the compiler doesn’t check this, it’s not smart enough to know this is a problem. So it’s not gonna force you to handle the exception, it just trusts that you know what you are doing. If you do it wrong, your app will crash.",
    "ts": "1472278461.000949",
    "id": "26be7416",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "I see",
    "ts": "1472278477.000950",
    "id": "26be7417",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "type out the code, run the example",
    "ts": "1472278507.000951",
    "id": "26be7418",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "best way to learn is to experience it",
    "ts": "1472278515.000952",
    "id": "26be7419",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "<@U08GFT5V5> yeah. thanks",
    "ts": "1472278533.000953",
    "id": "26be7420",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "And then, for a ‘checked’ exception, it’s more like, in Nish’s first example.. When you try to read in a file.\n\nThe compiler makes you handle a ‘checked’ exception. The compiler is saying “It’s possible an error will occur here. We know about this error. So you better make sure your code knows what to do when this error happens\"",
    "ts": "1472278539.000954",
    "id": "26be7421",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "alright, do you always have to throw an exception?",
    "ts": "1472278589.000955",
    "id": "26be7422",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "Checked exceptions usually come up when you are doing something where you are not in control of your own destiny so to speak",
    "ts": "1472278594.000956",
    "reactions": [
      {
        "name": "+1",
        "users": [
          "U25HPU938"
        ],
        "count": 1
      }
    ],
    "id": "26be7423",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "When you are reading a file, lots of stuff could go wrong",
    "ts": "1472278616.000957",
    "id": "26be7424",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "<@U0G47NBQV> what do you mean ‘do you always have to throw an exception’?",
    "ts": "1472278622.000958",
    "id": "26be7425",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "OS could delete the file",
    "ts": "1472278624.000959",
    "id": "26be7426",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "Data source could crap out, so you need to tell it what to do if something goes wrong",
    "ts": "1472278650.000960",
    "reactions": [
      {
        "name": "+1",
        "users": [
          "U0G47NBQV"
        ],
        "count": 1
      }
    ],
    "id": "26be7427",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "I mean, when catching the exception. can you do something other than throwing it like:\n``` String file = setName + \".dset\";\n    FileInputStream in = null;\n    try {\n        in = new FileInputStream(file);\n        return readDataSet(in);\n    } catch (IOException e) {\n    ;        // instead of “throw new getDataSetException()\"\n}\n```",
    "ts": "1472278755.000961",
    "id": "26be7428",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "Yeah you can do whatever you want",
    "ts": "1472278782.000962",
    "id": "26be7429",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "```\nString file = setName + \".dset\";\n    FileInputStream in = null;\n    try {\n        in = new FileInputStream(file);\n        return readDataSet(in);\n    } catch (IOException e) {\n        println(“Looks like someone f*cked something up…\");\n    }\n```",
    "ts": "1472278833.000963",
    "reactions": [
      {
        "name": "ok",
        "users": [
          "U0G47NBQV"
        ],
        "count": 1
      }
    ],
    "id": "26be7430",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "Yep. In fact, most of the time you probably won’t want to ‘re-throw’ the exception. You only do that when you actually want the app to crash when the exception is thrown.",
    "ts": "1472278834.000964",
    "id": "26be7431",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "Okay. I see. so every time you throw an exception the app mostly crashes?",
    "ts": "1472278909.000965",
    "id": "26be7432",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "```\nString file = setName + \".dset\";\n    FileInputStream in = null;\n    try {\n        in = new FileInputStream(file);\n        return readDataSet(in);\n    } catch (IOException e) {\n    \t//We're catching an exception, and we just want to log it or something.\n    \tLog.e(TAG, \"Something went wrong...\");\n    \t// But we still want the app to crash, because we just can't recover from this error. (rare)\n\t\tthrow new getDataSetException()\n}\n```\n\nOR\n\n```\nString file = setName + \".dset\";\n    FileInputStream in = null;\n    try {\n        in = new FileInputStream(file);\n        return readDataSet(in);\n    } catch (IOException e) {\n    \t//We're catching an exception, and we just want to log it or something.\n    \tLog.e(TAG, \"Something went wrong...\");\n    \t// Ah well, we crashed. Let's just carry on as if nothing happened.\n}\n```",
    "ts": "1472278959.000966",
    "id": "26be7433",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "Okay. right. i get it all now.",
    "ts": "1472278992.000967",
    "id": "26be7434",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": ":+1:",
    "ts": "1472279032.000968",
    "id": "26be7435",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "also, can you recover from an unchecked exception or not? or it will just report if something happens",
    "ts": "1472279077.000969",
    "id": "26be7436",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "Remember, unchecked exception is pretty much the same as a checked exception, the difference is that you must handle it, or your code won’t compile.",
    "ts": "1472279181.000970",
    "id": "26be7437",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "With that in mind, do you still need an answer to that question?",
    "ts": "1472279200.000971",
    "id": "26be7438",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "<@U0G47NBQV> runnable example for you",
    "ts": "1472279268.000972",
    "id": "26be7439",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": "<https://glot.io/snippets/ehvnqx3tmn>",
    "attachments": [
      {
        "title": "Exception demo",
        "title_link": "https://glot.io/snippets/ehvnqx3tmn",
        "text": "class Main { public static void main(String[] args) { int[] mArr = new int[2]; try { //Something dangerous mArr[3] = 5; } catch (A",
        "fallback": "Exception demo",
        "thumb_url": "https://glot.io/static/img/java.svg.png?etag=B6D1aOeo",
        "from_url": "https://glot.io/snippets/ehvnqx3tmn",
        "thumb_width": 380,
        "thumb_height": 514,
        "service_icon": "https://glot.io/favicon.ico",
        "service_name": "glot.io",
        "id": 1
      }
    ],
    "ts": "1472279269.000973",
    "id": "26be7440",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "right. got it. thanks <@U25HPU938> and <@U08GFT5V5>",
    "ts": "1472279275.000975",
    "id": "26be7441",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "<@U0G47NBQV> no worries. Can you have a look at <@U08GFT5V5>’s example? They went to the effort of creating it for you, and it will probably help you to fully understand what’s going on.",
    "ts": "1472279316.000976",
    "id": "26be7442",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "yep already looking at it now.",
    "ts": "1472279332.000977",
    "id": "26be7443",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "Cool :slightly_smiling_face:",
    "ts": "1472279336.000978",
    "id": "26be7444",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U0G47NBQV",
    "text": "thanks <@U25HPU938> and <@U08GFT5V5> appreciate all the help :slightly_smiling_face:",
    "ts": "1472279373.000979",
    "id": "26be7445",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U25HPU938",
    "text": "No worries",
    "ts": "1472279384.000980",
    "id": "26be7446",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  },
  {
    "user": "U08GFT5V5",
    "text": ":partyparrot:",
    "ts": "1472279389.000981",
    "id": "26be7447",
    "thread": [
      {
        "position": "null",
        "number": 0
      }
    ],
    "label": [
      {
        "type": "null",
        "position": "null"
      }
    ]
  }
]